### ISSUE TITLE
exclude dist folder from dev branch

### ISSUE THREAD

**Issue by micuat** [hydra-synth/hydra#280]
* perhaps it's better to exclude the bundled files and the entire dist folder (or is it necessary for the server??)
* maybe better to add `build.outDir: "dist"` in vite.config.js to make it explicit

I always have confusion when switching between main and dev :sweat:

**Comment by ojack** [comment]
I think the dist is necessary for publishing. What is the problem that is
happening?

**Comment by micuat** [comment]
@ojack that's what I thought. But it got confusing when I was trying to merge dev into another branch (diverged from dev). I think it's better to build it on the server, instead, I guess? Or github actions + vite build and publish to a separate branch like [this](https://github.com/sitek94/vite-deploy-demo-custom-domain)? I can experiment with it

**Comment by ojack** [comment]
Trying to remember earlier conversations, but I think a big motivation for
including the /dist (in addition to the current deployment process) is that
people who have no experience with build tools or node installed can
download and use hydra locally, without having to do anything else.

I am also open to other ways of doing this! but dont want to remove /dist
until there are workable alternatives for both things.

re: merging-- is it possible to delete the dist files as you are merging,
and then regenerate once the merge has happened? this big of a restructure
shouldn't happen too often, so hopefully once the merge happens it won't be
too complicated.

**Comment by micuat** [comment]
yes, I remember - while I agree with your point making it accessible, it's true that those who may want to use the editor needs to go through some steps on the backend (setting up http server etc).

My main motivation is before dev (vite) replaces main (browserify), we should clean up the pipeline. Given these points, maybe I should reframe this issue into "add docs for developers". I see there's already commands in the readme, but for those who want to make minor edits to the code, it may be helpful to have a small intro to how vite works (which file it is reading) and if someone wants to use their own hydra-synth, how can they use it in their self-hosted editor.

**Comment by micuat** [comment]
@ojack as first step I made a draft in v2 doc - can I go ahead and merge?

https://github.com/hydra-synth/hydra-docs-v2/pull/3
